atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -697,6 +699,35 @@ private void joinWithRetry(
         });
   }
 
+  public CompletableFuture<Void> leave() {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+
+    threadContext.execute(
+        () -> {
+          final var leaving = cluster.getLocalMember();
+          final var receiver =
+              membershipService.getMembers().stream()
+                  .filter(member -> !member.id().equals(leaving.memberId()))
+                  .findAny()
+                  .orElseThrow()
+                  .id();
```

Comment:
:x: Is this correct? The request can be only handled by the members of this partition's replication group.

Reply:
On this note, if there is already a leader you can send the request to the leader directly. Since this is running in the raft context, we can find the leader easily. If there is no leader, try to sent the request to any follower.

Reply:
Also, this might be a problem in join request as well. The request must be sent to a member that has this partition.

Reply:
Fixed as discussed: `join` will take a list of member ids (usually provided by the `RaftPartitionServer`'s partition metadata), `leave` will send to the leader if know or any other active, remote member as provided by the current configuration in `RaftClusterContext`.